### PR TITLE
feat(blog): Blog post about thinking about metrics.

### PR DESCRIPTION
**Pull request by jxnl**
| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit 07f8bda5e001825964fdb2d9db272ee6a3056e96.  | 
|--------|--------|

### Summary:
This PR adds MathJax configuration for rendering mathematical notation, a new blog post discussing metrics for retrieval augmented generation, and updates the site configuration to support these changes.

**Key points**:
- Added a new JavaScript file `mathjax.js` for configuring MathJax.
- Added a new blog post 'Stop using LGTM@Few as a metric (Better RAG)'.
- Updated `mkdocs.yml` to include `mathjax.js` and configure `pymdownx.arithmatex`.
----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Review by leehanchung — COMMENTED** [review]
Think it would be good to cover or classify rank-aware vs rank unaware metrics.

For RAG purposes, rank might not have as much impact vs 10 blue links sites. And even when it does, it's a problem with the generative models.

File: docs/writing/posts/lgtmk.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+## Importance of Velocity
+
+How quickly you can get metrics and run tests determines the nature of how you iterate on your software. If you're looking at a metric that takes a long time to compute, you're going to be waiting a long time to iterate on your system. So do whatever it takes to make the test that you run and the metrics you build as fast as possible!
+
+**Example via RAG**
+
+- **Slow Metric**: Collocating human preferences and consulting domain experts.
+- **Still Slow Metric**: AI-generated metrics. When using something like GPT4, things can become very slow.
+- **Fast Metrics**: Accuracy, Precision, Recall, MRR, NDCG, are computationally cheap given the labels.
```
Comment (leehanchung) [inline docs/writing/posts/lgtmk.md:63]: for RAG, it's usually hard to collect rank-aware data. users might not click on the references/citation links at all.  so mrr and ndcg will be hard to infer.

File: docs/writing/posts/lgtmk.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    In the context of search, recall is the fraction of relevant documents retrieved. Now, this is somewhat theoretical since we typically don't know how many relevant results there are in the index. Also, it's much easier to measure if the retrieved results are relevant, which brings us to ...
+
+### Mean Average Precision (MAP) @ K
+
+Assesses the accuracy of the top K retrieved documents, ensuring the relevance and precision of retrieved content.
+
+**Formula for Standard Precision**
+
+$$
+Precision@K = \frac{\text{Number of relevant documents in top K}}{K}
```
Comment (leehanchung) [inline docs/writing/posts/lgtmk.md:133]: for equivalency/better context, it's probably more clear to express as

```tex
Precision@K = \frac{\text{Number of relevant documents in top K}}{Total number of retrieved documents, or K}
```

**Comment by jxnl** [comment]
> Think it would be good to cover or classify rank-aware vs rank unaware metrics.
> 
> For RAG purposes, rank might not have as much impact vs 10 blue links sites. And even when it does, it's a problem with the generative models.

I think a lot of people underestimate how a good rag application ends up being a pretty strong document search tool for many companies. I kind of described that when I talk about MRR versus NDCG.

File: docs/writing/posts/lgtmk.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+I strongly recommend you not focus too hard on generation from an LLM, And to stay focused on being able to provide the right context. You will be able to get a language model to be more robust and as language models improve, they will only get more resilient to irrelevant information. However, as you build out your business, this data set that you curate on relevancy will stay with you.
+
+## Thinking about the metrics @ K
+
+Now let's look at some metrics that we can use to evaluate the performance of our retrieval augmented generation system. The goal isn't to give a mathematical breakdown of these metrics, but instead give you a sense of what they mean and how they might be useful. And how I like to explain and interpret them at the limits.
+
+### Mean Average Recall (MAR) @ K
+
+Focuses on the system's capability to retrieve all relevant documents within the top K results, emphasizing the breadth of relevant information captured.
```
Comment (leehanchung) [inline docs/writing/posts/lgtmk.md:110]: probably should note that the total number of relevant documents is a known unknown, as the search is usually incomplete/time-bound in a large enough system

File: docs/writing/posts/lgtmk.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+    3. If it is, consider building a router to conditionally select one implementation over another.
+
+## Metrics lead to Business Outcomes
+
+All of these metrics must ultimately be in service of something else. By improving things like precision, recall, and relevancy, what we're really hoping to do is generate better results for the business. The question then you have to ask yourself is, "What does that actually improve?". Here are a couple of things that you might want to consider.
+
+1. **User Satisfaction**: Are users happy with the answers they're getting? Could be defined by Thumb Up/Down or NPS.
+2. **Engagement**: Are users coming back to the platform? Are they spending more time on the platform?
+3. **Conversion**: Are users buying more things? Are they clicking on more ads?
+4. **Retention**: Are users staying on the platform longer? Are they coming back more often? Do we want to improve time spent?
```
Comment (leehanchung) [inline docs/writing/posts/lgtmk.md:213]: reducing time spent on the platform?  or increase number of distinct queries per same unit of time measurement? as a proxy for productivity boost?

Comment (leehanchung) [reply docs/writing/posts/lgtmk.md:213]: the better the search the short time user will use the system.

**Review by leehanchung — COMMENTED** [review]
some other notes

**Comment by leehanchung** [comment]
> > Think it would be good to cover or classify rank-aware vs rank unaware metrics.
> > For RAG purposes, rank might not have as much impact vs 10 blue links sites. And even when it does, it's a problem with the generative models.
> 
> I think a lot of people underestimate how a good rag application ends up being a pretty strong document search tool for many companies. I kind of described that when I talk about MRR versus NDCG.

i think its the opposite.  existing document search tool with an established query log got a huge edge in rag.

document search is a whole different ball game for many companies w/ permissions, e.g., sharepoint.
